tests/resources/functional/tsql/cte/multiple_cte.sql

```diff
@@ -0,0 +1,19 @@
+--
+-- Verify a few CTEs that include multiple expressions.
+--
+
+-- tsql sql:
+WITH a AS (SELECT 1, 2, 3),
+     b AS (SELECT 4, 5, 6),
+     c AS (SELECT * FROM a)
+SELECT * from b
+UNION
+SELECT * FROM c;
+
+-- databricks sql:
+WITH a AS (SELECT 1, 2, 3),
+     b AS (SELECT 4, 5, 6),
+     c AS (SELECT * FROM a)
+(SELECT * from b)
```

Comment:
I'm not sure the parenthesis are required here

Reply:
Same situation as [here](https://github.com/databrickslabs/remorph/pull/1273/files/38fc3b3c0c96a2c654d2af20a42381ea2ff6e70c#r1869060713).